packages/inngest/src/components/InngestStepTools.ts

```diff
@@ -683,6 +749,26 @@ export const createStepTools = <TClient extends Inngest.Any>(
      * this context, and a custom fallback can be set using the `config` method.
      */
     fetch: stepFetch,
+
+    /**
+     * Create a durable metadata update wrapped in a step
+     *
+     * @param memoizationId - The step ID used for the step itself, ensuring the
+     *   metadata update is only performed once even on function retries.
+     *
+     * @example
+     * ```ts
+     * // Update metadata for the current run
+     * await step.metadata("update-status").update({ status: "processing" });
+     *
+     * // Update metadata for a different run
+     * await step.metadata("notify-parent")
+     *   .run(parentRunId)
+     *   .update({ childCompleted: true });
+     * ```
+     */
+    metadata: (memoizationId: string): MetadataStepTool =>
+      createStepMetadataWrapper(memoizationId),
```

Comment:
Fantastic! Did what you needed to do here make sense? I'd quite like us to move to this in v4 🤔